workers/unity/Packages/com.improbable.gdk.core/GameObjectRepresentation/Injection/InjectableId.cs

```diff
@@ -0,0 +1,33 @@
+﻿using System;
+
+namespace Improbable.Gdk.Core.GameObjectRepresentation
+{
+    /// <summary>
+    ///     Enum to note what kind of Injectable a specific type is, to be used as part of the InjectableId.
+    /// </summary>
+    public enum InjectableType
+    {
+        ReaderWriter,
+        CommandSender,
+        CommandRequestHandler,
+        CommandResponseHandler
+    }
+
+    /// <summary>
+    ///     Identifier for specific injectable type, to be used for finding IInjectableCreators and querying from
+    ///     the InjectableStore.
+    /// </summary>
+    public struct InjectableId
+    {
+        public readonly InjectableType type;
+        public readonly uint componentId;
+        public readonly int index;
```

Comment:
What is this for? Is this the command index of a command of a component?

Reply:
Command index, didn't want to name it that in case we want to use it for any other purpose in other types later... Might not be the best way to go about this, what do you think?

Reply:
Its rather unclear looking just at this object. Do we expect do have other injectable objects with extra information?

Reply:
Not really, and we could change it if we do add extra info. It just felt weird specifying a field to a use case of a subset of these objects, but I guess it's still better than a random thing just hanging there like that?

Reply:
Without understanding the implications of this decision to its full extent, my general opinion would be to not overload the use of a field unless necessary. Since that's not even the case right now and `index` is only ever used for command indices, let's make it unambiguous and rename it to `commandIndex`. That makes the intent clearer. We can also still refactor this later if it ever becomes necessary.